core/os/memory.cpp

```diff
@@ -43,41 +44,36 @@ void *operator new(size_t p_size, void *(*p_allocfunc)(size_t p_size)) {
 	return p_allocfunc(p_size);
 }
 
-#include <stdio.h>
-
 #ifdef DEBUG_ENABLED
 size_t Memory::mem_usage = 0;
 size_t Memory::max_usage = 0;
+#define PREPAD true
+#else
+#define PREPAD p_pad_align
```

Comment:
Or does it get replaced "as is" and it gets evaluated properly as it's by chance only used in the scope where `p_pad_align` exists? In this case, I have the feeling that the previous code with a local boolean was clearer and less error-prone, but maybe this change improves performance?

(BTW those are genuine questions from a less-than-average C++ dev, so feel free to tell me if I'm wrong because I don't understand an underlying concept :D).